it/lessons/basics/functions.md

```diff
@@ -90,6 +91,195 @@ iex> Length.of [1, 2, 3]
 3
 ```
 
+### Nomi di funzioni e _Arity_
+
+Abbiamo accennato in precedenza che le funzioni vengono definite dal nome della funzione e dalla _arity_ (numero di parametri della funzione).
+Questo significa che possiamo fare:
+
+```elixir
+defmodule Greeter2 do
+  def hello(), do: "Hello, anonymous person!"   # hello/0
+  def hello(name), do: "Hello, " <> name        # hello/1
+  def hello(name1, name2), do: "Hello, #{name1} and #{name2}"
+                                                # hello/2
+end
+
+iex> Greeter2.hello()
+"Hello, anonymous person!"
+iex> Greeter2.hello("Fred")
+"Hello, Fred"
+iex> Greeter2.hello("Fred", "Jane")
+"Hello, Fred and Jane"
+```
+
+Abbiamo elencato i nomi delle funzioni nei commenti accanto ad ogni funzione.
+La prima implementazione non accetta parametri, quindi si chiama `hello/0`; la seconda accetta un parametro, quindi si chiama `hello/1`, e così via.
+A differenza del _function overloading_ in altri linguaggi di programmazione, in Elixir, queste sono funzioni _diverse_ l'una dall'altra.
+(Pattern matching, descritta qui sopra, può essere applicata solo quando vengono fornite piú definizioni di funzioni con lo _stesso_ numero di parametri.)
+
+### Funzioni e Pattern Matching
+
+Dietro le quinte, le funzioni eseguiscono automaticatemente il _pattern-matching_ sui parametri con cui vengono chiamate.
+
+Per esempio, abbiamo bisogno di una funzione che accetti una _map_ ma siamo interessati solo in una chiave in particolare; possiamo applicare il _patter-match_ sul parametro così:
+
+```elixir
+defmodule Greeter1 do
+  def hello(%{name: person_name}) do
+    IO.puts "Hello, " <> person_name
+  end
+end
+```
+
+E se abbiamo una _map_ che descrive una persona chiamata Fred:
+
+```elixir
+iex> fred = %{
+...> name: "Fred",
+...> age: "95",
+...> favorite_color: "Taupe"
+...> }
+```
+
+Questo è il risultato che otterremo quando chiamiamo `Greeter1.hello/1` con la mappa `fred`.
+
+```elixir
+# chiamata con l'intera mappa
+...> Greeter1.hello(fred)
+"Hello, Fred"
+```
+
+Cosa succede quando chiamiamo la funzione con una mappa che _non_ contiente la chiave `:name`?
+
+```elixir
+# call without the key we need returns an error
+...> Greeter1.hello(%{age: "95", favorite_color: "Taupe"})
+** (FunctionClauseError) no function clause matching in Greeter1.hello/1
+
+    The following arguments were given to Greeter1.hello/1:
+
+        # 1
+        %{age: "95"}
+
+    iex:12: Greeter1.hello/1
+
+```
+
+La ragione di questo comportamento è che Elixir eseguise il _patter-matching_ dei parametri con cui la funzione è stata chiamata contro la funzione della stessa _arity_.
```

Comment:
* `eseguise` should be `esegue`
* typo in `patter-matching`

Reply:
fixed :)